### PR TITLE
feat: simply state loop in sc & fix typo

### PR DESCRIPTION
**Pull request by hunjixin**
### Review Type Requested (choose one):
- [ ] **Glance** - superficial check (from domain experts)
- [ ] **Logic** - thorough check (from everybody doing review)

this pr is pick pow smart  constract's changes from https://github.com/Lilypad-Tech/lilypad/pull/163

1. fix typo
2. simplfy the way to get state in sc
3. remove a not nessasry api function

### Summary
Provide a one line summary and link to any relevant references

### Task/Issue reference

Closes: add_link_here

### Details (optional)
Add any additional details that might help Code Reviewers digest this PR

### CODE REVIEW

File: hardhat/contracts/LilypadPow.sol [hunk 1]
```diff
@@ -45,10 +44,14 @@ contract LilypadPow is Ownable, Initializable {
     // https://docs.openzeppelin.com/upgrades-plugins/1.x/writing-upgradeable
     function initialize() public initializer {}
 
-    function getMiners() external view returns (address[] memory) {
+    function getMiners() public view returns (address[] memory) {
         return miners;
     }
 
+   function getMinerPosSubmissions(address addr) public view returns (POWSubmission[] memory) {
```
Comment (richardbremner) [inline hardhat/contracts/LilypadPow.sol]: I think there's a typo here Pos > Pow

**Review by richardbremner — COMMENTED** [review]
shall we add the purge function as discussed?

File: hardhat/contracts/LilypadPow.sol (hunk 1 above)
Comment (hunjixin) [reply hardhat/contracts/LilypadPow.sol]: yes，you are right, wait for minutes.

**Comment by hunjixin** [comment]
> shall we add the purge function as discussed?

i can add it in another pr if nessary. we can merge this one first.

**Comment by hunjixin** [comment]
@richardbremner  can you review again

**Review by richardbremner — APPROVED** [review]
approval the content but lets make sure we know how we are releasing the sc changes before doing so

**Pull request merged**